src/components/organisms/Modal/Modal.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  FC,
+  ReactNode,
+  PropsWithChildren,
+  Dispatch,
+  SetStateAction,
+} from 'react'
+import * as Dialog from '@radix-ui/react-dialog'
+import classNames from 'classnames'
+import Button, { AppearanceTypes } from 'components/molecules/Button/Button'
+import { ReactComponent as Close } from 'assets/icons/close.svg'
+import { useTranslation } from 'react-i18next'
+
+import classes from './styles.module.scss'
+
+export enum ModalSizeTypes {
+  Narrow = 'narrow',
+  Medium = 'medium',
+  Big = 'big',
+}
+
+export enum ButtonPositionTypes {
+  SpaceBetween = 'spaceBetween',
+  Right = 'right',
+  None = 'none',
+}
+
+export enum TitleFontTypes {
+  Gray = 'gray',
+  Black = 'black',
+}
+
+export interface ModalProps {
+  title?: string
+  trigger?: ReactNode
+  customDialogContent?: string
+  topButton?: boolean
+  size?: ModalSizeTypes
+  buttonsPosition: ButtonPositionTypes
+  breakButtonLabel?: string
+  proceedButtonLabel?: string
+  handleClose: () => void
+  handleProceed: () => void
+  open: boolean
+  setOpen: Dispatch<SetStateAction<boolean>>
+  titleFont?: TitleFontTypes
+}
+
+export interface ModalFooterProps {
+  buttonsPosition: ButtonPositionTypes
+  breakButtonLabel?: string
+  proceedButtonLabel?: string
+  handleClose: () => void
+  handleProceed: () => void
+}
+
+const ModalFooter: FC<ModalFooterProps> = ({
+  buttonsPosition = 'right',
+  breakButtonLabel,
+  proceedButtonLabel,
+  handleClose,
+  handleProceed,
+}) => {
+  return (
+    <div className={classes[buttonsPosition]}>
+      <Button appearance={AppearanceTypes.Secondary} onClick={handleClose}>
+        {breakButtonLabel}
+      </Button>
+      <Button className={classes.modalButton} onClick={handleProceed}>
+        {proceedButtonLabel}
+      </Button>
+    </div>
+  )
+}
+
+const Modal: FC<PropsWithChildren<ModalProps>> = ({
+  title,
+  trigger,
+  children,
+  topButton,
+  titleFont = 'gray',
+  size = 'medium',
+  buttonsPosition,
+  breakButtonLabel,
+  proceedButtonLabel,
+  open,
+  setOpen,
+  handleClose,
+  handleProceed,
+}) => {
+  const { t } = useTranslation()
+
+  return (
+    <Dialog.Root open={open} onOpenChange={setOpen}>
+      {trigger}
+      <Dialog.Portal>
+        <Dialog.Overlay className={classes.dialogOverlay} />
+        <Dialog.Content
+          className={classNames(classes.dialogContent, classes[size])}
+        >
+          {topButton && (
+            <Button
+              icon={Close}
+              appearance={AppearanceTypes.Secondary}
+              onClick={handleClose}
+              className={classes.topButton}
+            >
+              {t('button.cancel')}
+            </Button>
+          )}
+          <Dialog.Title
```

Comment:
**[blocker]**

Currently we only support having a string title with 2 different font styles and it is always at the top of the modal, however we have cases, where instead of the title we have some other component. Like the progress bar here: 

https://www.figma.com/file/VqCwED3Dymn7L1gGoAOK27/EKI-t%C3%B5lkev%C3%A4rav?type=design&node-id=834-80371&t=Z4cUh8GLdliLnTqX-4